providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/file_watcher.py

```diff
@@ -0,0 +1,59 @@
+import json
+import logging
+import os
+import re
+import threading
+import time
+import typing
+
+from openfeature.event import ProviderEventDetails
+from openfeature.provider.provider import AbstractProvider
+
+
+class FileWatcherFlagStore:
+    def __init__(self, file_path: str, provider: AbstractProvider):
+        self.file_path = file_path
+        self.provider = provider
+        self.last_modified = 0.0
+        self.load_data()
+        self.thread = threading.Thread(target=self.refresh_file, daemon=True)
+        self.thread.start()
+
+    def shutdown(self) -> None:
+        pass
+
+    def get_flag(self, key: str) -> typing.Optional[dict]:
+        return self.flag_data.get(key)
+
+    def refresh_file(self) -> None:
+        while True:
+            time.sleep(1)
+            logging.debug("checking for new flag store contents from file")
+            last_modified = os.path.getmtime(self.file_path)
+            if last_modified > self.last_modified:
+                self.load_data(last_modified)
+
+    def load_data(self, modified_time: typing.Optional[float] = None) -> None:
+        # TODO: error handling
```

Comment:
Please add error handling or create a follow-up issue.

Reply:
Put in some proper error handling for file reads, and also moved the logic to another class so it is run at parse time and not evaluation time. Feel free to suggest improvements to the new structure

Reply:
I think it would also be a good idea to include the schema via submodule and parse the verify the data against that explicitly at parse time

Reply:
Do you want to include that in the PR or should we create a follow up issue?

Reply:
Follow-up issue sounds good, not sure how simple it would be to introduce